### ISSUE TITLE
Create a C# application to handle creating a `NovelRT` project

### ISSUE THREAD

**Issue by capnkenny** [novelrt/NovelRT#323]
We will want to create a C# application (.NET Core / 5 ) that will be able to handle the following:

- Pulling down NovelRT and the relevant dependencies from `conan`
- Generating a new CMake project that points to the proper dependencies and the Engine/Interop libs that were pulled down

- Generating a .NET project in the appropriate location w/ the .NET support libraries and Ink libraries included

**Comment by capnkenny** [comment]
@RubyNova - would we need to publish the C# bindings as a NuGet package first for the .NET project to be able to reference them? Or would we rather build it to reference as a local dependency? (Same with Ink, as it's already hosted as a package IIRC)

**Comment by RubyNova** [comment]
This sounds like a question for @tannergooding but in my mind I think nuget would be easiest? Not sure though.

**Comment by thaumanovic** [comment]
Is an application the best idea? Are build templates something that would be able to accomplish this?

**Comment by RubyNova** [comment]
No. Build templates assume everything is C# driven, when it isn't - it's all CMake driven. Not only that but since the project is multi-language I imagine strictly relying on MSBuild from a csproj file, when .NET itself is optional to include in your project, might become messy.

**Comment by RubyNova** [comment]
@capnkenny how has this changed now we have changed the design philosophy post-MVP?

**Comment by RubyNova** [comment]
I'm unsure what the status on my question is. bump.

**Comment by capnkenny** [comment]
I'm not sure what is meant by how has this changed - you mean the objective of the SDK? How C# is included?
